### PR TITLE
feat: expose constraint metamodel in Quarkus and Spring Boot

### PR DESCRIPTION
**Pull request by triceo**

### CODE REVIEW

File: core/src/main/java/ai/timefold/solver/core/api/score/stream/ConstraintBuilder.java [hunk 1]
```diff
@@ -20,12 +21,26 @@ default Constraint asConstraint(String constraintName) {
...
+        return asConstraintDescribed(constraintName, constraintDescription, Constraint.DEFAULT_CONSTRAINT_GROUP);
+    }
+
+    /**
+     * Builds a {@link Constraint} from the constraint stream.
+     * The {@link ConstraintRef#packageName() constraint package} defaults to the package of the {@link PlanningSolution} class.
+     *
+     * @param constraintName never null, shows up in {@link ConstraintMatchTotal} during score justification
+     * @param constraintDescription never null
+     * @param constraintGroup never null, only allows alphanumeric characters, "-" and "_"
```
Comment (zepfred) [inline core/src/main/java/ai/timefold/solver/core/api/score/stream/ConstraintBuilder.java:40]: The regex is using `\p{So}`, which means we will accept other symbols.

Comment (triceo) [reply core/src/main/java/ai/timefold/solver/core/api/score/stream/ConstraintBuilder.java:40]: True. Geoffrey and I thought it'd be nice to allow Emoji, for demo purposes. At the same time, does it need to be documented?

File: core/src/test/java/ai/timefold/solver/core/impl/score/director/stream/DefaultConstraintMetaModelTest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+class DefaultConstraintMetaModelTest {
+
+    @Test
+    void test() {
+        var solutionDescriptor = TestdataSolution.buildSolutionDescriptor();
+        var constraintFactory = new TestConstraintFactory<TestdataSolution, SimpleScore>(solutionDescriptor);
+        var constraint1 = new TestConstraint<>(constraintFactory, "Test Constraint 1", SimpleScore.of(1));
+        var constraint2 = new TestConstraint<>(constraintFactory, "Test Constraint 2", SimpleScore.of(10));
+        var constraint3 = new TestConstraint<>(constraintFactory, "Test Constraint 3", "test", SimpleScore.of(100));
+        var constraint4 = new TestConstraint<>(constraintFactory, "Test Constraint 4", "another-test", SimpleScore.ZERO);
```
Comment (zepfred) [inline core/src/test/java/ai/timefold/solver/core/impl/score/director/stream/DefaultConstraintMetaModelTest.java:25]: This is a valid name.

```
        var constraint5 = new TestConstraint<>(constraintFactory, "Test Constraint 5", "group⏺", SimpleScore.of(1));

```

File: core/src/test/java/ai/timefold/solver/core/impl/score/stream/common/inliner/AbstractScoreInlinerTest.java [hunk 3]
```diff
@@ -35,54 +32,6 @@ protected TestConstraint<Solution_, Score_> buildConstraint(Score_ constraintWei
...
-    public static final class TestConstraint<Solution_, Score_ extends Score<Score_>>
-            extends AbstractConstraint<Solution_, TestConstraint<Solution_, Score_>, TestConstraintFactory<Solution_, Score_>> {
-
-        protected TestConstraint(TestConstraintFactory<Solution_, Score_> constraintFactory, String constraintName,
-                Score_ constraintWeight) {
-            super(constraintFactory, ConstraintRef.of(constraintFactory.getDefaultConstraintPackage(), constraintName), "",
-                    constraintWeight, ScoreImpactType.REWARD, null, null);
-        }
-    }
+    ;
```
Comment (zepfred) [inline core/src/test/java/ai/timefold/solver/core/impl/score/stream/common/inliner/AbstractScoreInlinerTest.java:35]:
```suggestion
```

File: core/src/main/java/ai/timefold/solver/core/api/score/stream/ConstraintBuilder.java (hunk 1 above)
Comment (zepfred) [reply core/src/main/java/ai/timefold/solver/core/api/score/stream/ConstraintBuilder.java:40]: Just reading the documentation, I wouldn't assume that we accept these symbols.

File: quarkus-integration/quarkus/deployment/src/test/java/ai/timefold/solver/quarkus/TimefoldProcessorConstraintMetaModelTest.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+import org.assertj.core.api.Assertions;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+
+class TimefoldProcessorConstraintMetaModelTest {
```
Comment (zepfred) [inline quarkus-integration/quarkus/deployment/src/test/java/ai/timefold/solver/quarkus/TimefoldProcessorConstraintMetaModelTest.java:18]: Please also update `TimefoldProcessorSolverResourcesTest`

**Review by zepfred — DISMISSED** [review]
LGTM!

**Comment by zepfred** [comment]
Please address the SonarCloud issues

**Review by zepfred — APPROVED** [review]

**Pull request merged**